### DISCUSSION TITLE
Hide mouse cursor

### DISCUSSION THREAD

**Discussion by garentyler** [bevyengine/bevy#3292] [Q&A] [answered]
I would like to hide the mouse cursor, but I can't find anything about it.

**Comment by rparrett** [comment, marked as answer]
Here's the relevant docs, and an example

https://docs.rs/bevy/0.5.0/bevy/window/struct.Window.html#method.set_cursor_visibility
https://bevy-cheatbook.github.io/cookbook/mouse-grab.html

**Comment by jakobhellermann** [comment]
You can call `window.set_cursor_visibility(false)` for the primary window:
```rust
fn hide_cursor(mut windows: ResMut<Windows>) {
    let window = windows.get_primary_mut().unwrap();
    // window.set_cursor_lock_mode(true);
    window.set_cursor_visibility(false);
}
```

**Comment by garentyler** [comment]
Thank you!

**Comment by AnAverageGitUser** [comment]
In bevy 0.11 you have to set a property on the windows "cursor" object:
```
pub fn hide_cursor(
    mut primary_window: Query<&mut Window, With<PrimaryWindow>>,
) {
    let mut window = &mut primary_window.single_mut();
    window.cursor.visible = false;
}
```
https://docs.rs/bevy/0.11.3/bevy/window/struct.Cursor.html

**Comment by rip3rs** [comment]
As of Bevy 0.15 it is via the CursorOptions

```rust
pub fn hide_cursor(
  mut cursor: Single<&mut CursorOptions>,
) {
  cursor.visible = false;
}
``` 
https://docs.rs/bevy/0.15.0/bevy/window/struct.CursorOptions.html
